cargo-creusot/src/why3_launcher.rs

```diff
@@ -0,0 +1,124 @@
+use std::{
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use super::helpers::Result;
+use anyhow::anyhow;
+use include_dir::{include_dir, Dir};
+
+static PRELUDE: Dir<'static> = include_dir!("$CARGO_MANIFEST_DIR/../prelude");
+
+#[derive(Copy, Clone, Debug)]
+pub enum Why3Mode {
+    Ide,
+}
+
+impl std::fmt::Display for Why3Mode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Why3Mode::Ide => f.write_str("ide"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Why3Launcher {
+    mode: Why3Mode,
+    why3_path: Option<PathBuf>,
```

Comment:
Given that we cannot launch why3 (according to line 47) without this, can we make this not optional? To me, the error should be launched in the builder instead.

Reply:
Ok, like you want